### PR TITLE
Security stuff, add userId to Comment

### PR DESCRIPTION
**Pull request by hisahi**
### Short description
#826
Also removes unused backend code

### DoD

I have:
- [x] added actual code.
- [x] produced clean code that passes ESLint.
- [x] code that actually does what it should.
- [ ] documented the code or added documentation to the wiki.
- [ ] added or modified test(s).
- [x] test(s) that actually pass.

### CODE REVIEW

File: backend/server/controllers/courseinstances.js [hunk 1]
```diff
@@ -551,7 +584,7 @@ module.exports = {
                 weekMaxPoints: req.body.weekMaxPoints || courseInstance.weekMaxPoints,
                 currentWeek: req.body.currentWeek || courseInstance.currentWeek,
                 finalReview: req.body.finalReview,
-                currentCodeReview: req.body.newCr.length === 0 ? '{}' : `{${req.body.newCr.join(',')}}`,
+                currentCodeReview: newCr.length === 0 ? [] : `{${newCr.join(',')}}`,
```
Comment (yumoL) [inline backend/server/controllers/courseinstances.js]: curretCodeReview should be an array but `{${newCr.join(',')}}` doesn't seem to be an array

Comment (gotonode) [reply backend/server/controllers/courseinstances.js]: Is this change because we updated the addon?

**Comment by gotonode** [comment]
Looks good. Your velocity is astonishing. It leaves me to shame.

File: backend/server/controllers/courseinstances.js (hunk 1 above)
Comment (hisahi) [reply backend/server/controllers/courseinstances.js]: Using an array caused a postgres error in Sequelize v5

**Pull request merged**